sharing/README.md

```diff
@@ -25,3 +25,172 @@ High level features include:
+* **Categorical Type**: For categorical entities, this column should specify the category values to filter out seperated one again by the **;** symbol
+
+Users can also use the **all** keyword to filter our all values in an entity.
+
+For the data type, please refer to the [ODM metadata](https://github.com/Big-Life-Lab/ODM/blob/main/metadata_en.md#entity-relationship-diagram) on the allowable value for each one.
+
+Some examples are given below,
+
+1. Exclude values of `rawWW` or `swrSed`
+
+    | filterValue             |
+    |-------------------------|
+    | rawWW;swrSed            |
+
+2. Exclude values between March 1st 2021 and December 1st 2021 including the endpoints
+
+    | filterValue             |
+    |-------------------------|
+    | [2021-03-01,2021-12-01] |
+
+## Example Scenarios
+
+In this section we will be working with some data, providing an example scenario for a rule and showing what the rule looks like
+
+The data we will be working with has two tables from the ODM, **Sample** and **Site**. It does not include all the columns present in these tables. The rows in the Sample and Site table respectively are shown below:
+
+| sampleID | siteID | dateTime   | type    | sizeL | notes  | 
+|----------|--------|------------|---------|-------|--------|
+| 1        | 1      | 2021-08-19 | rawWW   | 5     | Note 1 |
+| 2        | 1      | 2021-08-18 | sweSed  | 3     | Note 2 |
+| 3        | 2      | 2021-08-17 | pstGrit | 15    | Note 3 |
+| 4        | 2      | 2020-01-10 | water   | 45    | Note 4 |
+
+| siteID | name                 | publicHealthDepartment | type   |
+|--------|----------------------|------------------------|--------|
+| 1      | University of Ottawa | Ottawa Public Health   | school |
+| 1      | University of Laval  | Laval Public Health    | school |
+
+
+1. Remove rows whose site ID in the Sample table is 2 for Ottawa Public Health (OPH)
+
+    | sharedWith | tableName | variableName | direction | filterValue |
+    |------------|-----------|--------------|-----------|-------------|
+    | OPH        | Sample    | siteID       | row       | 1           |
+
+2. Remove rows from the Sample table whose type is `rawWW` or `sweSed` for the Public Health Agency of Canada (PHAC)
+
+    | sharedWith | tableName | variableName | direction | filterValue  |
+    |------------|-----------|--------------|-----------|--------------|
+    | PHAC       | Sample    | type         | row       | rawWW;sweSed |
+
+3. Remove the type column from all tables for Laval Public Health (LPH)
+
+    | sharedWith | tableName | variableName | direction | filterValue  |
+    |------------|-----------|--------------|-----------|--------------|
+    | LPH        | all       | type         | column    | all          |
+
+4. Remove samples taken in the year 2021 and those whose volume is less than 5 for OPH and PHAC
```

Comment:
#4 will require more discussion. We will need a process of linking the two sharing rules together (the two rows in the sharing table). Also what if this is an 'or' filter?

Reply:
Here is an example that would be ideal to accommodate:

For Ottawa, exclude rows and columns in WWMeasure  in March 2021 ANDS have [`qualityFlag` = False  OR LOD < 100]. Exclude all other rows and columns in other tables during the same period with uuID that were excluded in the above rule.

This may be too tricky a rule to include, but it is a real example from Ottawa during the spring where there the spring melt. The Ottawa lab may not what to include the results from that melt period where the testing quality was poor.

Reply:
The validation approach in the Cerebrus package can accommodate this type of complex rule. Right?

Another typical approach that users are familiar with is numbering each row in share.csv and then create a rule with AND and OR. 

rule1 AND [rule2 or rule3].

As a general rule, we may want to add a unique ID to the rule `ruleID`. This allows people to more easily reference, duplicate, update, share rules. The ruleID would then facilitate more complicated sharing schema.

Reply:
This was a mistake, the example should be "Remove samples taken in the year 2021 **or** those whose volume is less than 5 for OPH and PHAC". Two different rules are implicitly combined using the OR operator and not AND.

With regards to your other points, I agree, we need to think of a solution "combine" rules together if we want to accommodate all scenarios. I was thinking along the same lines as your previous point but I'll address in a new PR since it would be an addon.

I'll add in a **ruleId** column.